src/sinks/new_relic_logs.rs

```diff
@@ -62,13 +56,15 @@ inventory::submit! {
     SinkDescription::new::<NewRelicLogsConfig>("new_relic_logs")
 }
 
-impl_generate_config_from_default!(NewRelicLogsConfig);
+impl GenerateConfig for NewRelicLogsConfig {
+    fn generate_config() -> toml::Value {
+        toml::Value::try_from(Self::with_encoding(Encoding::Json)).unwrap()
+    }
+}
 
-#[derive(Deserialize, Serialize, Debug, Eq, PartialEq, Clone, Derivative)]
+#[derive(Deserialize, Serialize, Debug, Eq, PartialEq, Clone)]
 #[serde(rename_all = "snake_case")]
-#[derivative(Default)]
 pub enum Encoding {
-    #[derivative(Default)]
     Json,
```

Comment:
I think we should add `Text` here as an encoding option.

Reply:
Are you okay with adding in another PR? This will not be breaking change because we already removed `Default` for `Encoding`.

Reply:
:+1: a separate PR is fine, but it'd be nice to do it before this change is released in a new version as it'd be confusing to require users to specify the `encoding.codec` option when there is only one valid value.